sdk/dotnet/Pulumi.Automation.Tests/LocalWorkspaceTests.cs

```diff
@@ -939,5 +937,32 @@ public void ValidVersionTheory(string currentVersion, bool errorExpected)
                 LocalWorkspace.ValidatePulumiVersion(testMinVersion, currentVersion);
             }
         }
+
+        [Fact]
+        public async Task RespectsProjectSettingsTest()
+        {
+            var program = PulumiFn.Create<ValidStack>();
+
+            var stackName = $"{RandomStackName()}";
+            var projectName = "project_was_overwritten";
+
+            var workdir = ResourcePath(Path.Combine("Data", "correct_project"));
+
+            var stack = await LocalWorkspace.CreateStackAsync(
+                new InlineProgramArgs(projectName, stackName, program)
+                {
+                    WorkDir = workdir
```

Comment:
The constructor of InlineProgramArgs crates default ProjectSettings implicitly here, without user intending to customize them.